store/postgres/src/deployment_store.rs

```diff
@@ -729,14 +729,16 @@ impl DeploymentStore {
 
                         cancel.check_cancel()?;
 
+                        let subgraph_error_count = deployment::error_count(conn, &site.deployment)?;
+
                         // FIXME: (Determinism)
                         //
                         // It is vital to ensure that the block hash given in the query
                         // is a parent of the latest block indexed for the subgraph.
                         // Unfortunately the machinery needed to do this is not yet in place.
                         // The best we can do right now is just to make sure that the block number
                         // is high enough.
-                        if latest_block_ptr.number < block.number {
+                        if latest_block_ptr.number < block.number && subgraph_error_count != 0 {
```

Comment:
We should be returning a PoI when there is an error. I though that was the whole point of this PR. What needs to be verified here is whether the error is a deterministic one.

Also let's fix this determinism issue (above). Determinism is non-negotiable.

Reply:
The change to the `instance_manager` so that the PoI entity get's saved to the database is enough for the GraphQL query to return the PoI for the failed block.

I think I made the `error_count` change because of some discussions I had at the time about fetching the PoI for future blocks. I think this can be removed.

About the determinism issue above, I'll don't know if we have the "machinery" to fix it right now, but I'll definitely investigate 🙂

Reply:
I don't think the check can be removed entirely, we still need to verify that the errors are deterministic before producing a PoI... unless non-deterministic errors do not `increment latest_block_ptr.number`?

Reply:
The `store.transact_block_operations` that happens in the `instance_manager` does exactly that. This PR makes the call for that function happen so:

- The PoI gets saved to the database even when deterministic errors happen
- The latest_block_ptr gets updated (makes the GraphQL PoI fetch work)

From my understanding this `if` here should work like a confidence check. So (if possible) we should make it stop comparing block number values, and actually check if the block number refers to the correct parent hash.

Reply:
Removed the `error_count` part: https://github.com/graphprotocol/graph-node/pull/2748/commits/7c06e32bdaede435d77f1d92ef74a6881eed4dcd

I'll create an issue for this and talk with David about it.